oaknational/Oak-Web-Application
# Added in `<FocusWrap/>` component so we can deal with complex focus logic

## Description
Adds `<FocusWrap/>` component so we can deal with complex focus logic and movement across multiple modals.

This allows us to wrap some focusable elements with `<FocusWrap>{/*...*/}</FocusWrap>` and react to "focus wrapping" events.

This PR also adds the focus logic as described in `CUR-878`

@rightsaidjames are you ok to do an early smoke test for a11y issues? 

## Issue(s)

Fixes `CUR-878`

## How to test

1. Go to https://deploy-preview-2926--oak-web-application.netlify.thenational.academy/teachers/curriculum
2. Check tab behaviour is correct as per designs in `CUR-878`
3. Check for any accessibility issues created by the designs
